File: chatwoot/java/src/main/java/oseg/chatwoot_examples/UpdateConversationExample.java
package oseg.chatwoot_examples;

import com.chatwoot.client.ApiException;
import com.chatwoot.client.Configuration;
import com.chatwoot.client.api.*;
import com.chatwoot.client.auth.*;
import com.chatwoot.client.JSON;
import com.chatwoot.client.model.*;

import java.io.File;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class UpdateConversationExample
{
    public static void main(String[] args)
    {
        var config = Configuration.getDefaultApiClient();
        ((ApiKeyAuth) config.getAuthentication("userApiKey")).setApiKey("USER_API_KEY");
        // ((ApiKeyAuth) config.getAuthentication("agentBotApiKey")).setApiKey("AGENT_BOT_API_KEY");

        var updateConversationRequest = new UpdateConversationRequest();

        try
        {
            new ConversationsApi(config).updateConversation(
                0, // accountId
                0, // conversationId
                updateConversationRequest // data
            );
        } catch (ApiException e) {
            System.err.println("Exception when calling ConversationsApi#updateConversation");
            System.err.println("Status code: " + e.getCode());
            System.err.println("Reason: " + e.getResponseBody());
            System.err.println("Response headers: " + e.getResponseHeaders());
            e.printStackTrace();
        }
    }
}

File: chatwoot/kotlin/src/main/kotlin/oseg/chatwoot_examples/UpdateConversationExample.kt
package oseg.chatwoot_examples

import com.chatwoot.client.infrastructure.*
import com.chatwoot.client.apis.*
import com.chatwoot.client.models.*

import java.io.File
import java.time.LocalDate
import java.time.OffsetDateTime
import kotlin.collections.ArrayList
import kotlin.collections.List
import kotlin.collections.Map
import com.squareup.moshi.adapter

@ExperimentalStdlibApi
class UpdateConversationExample
{
    fun updateConversation()
    {
        ApiClient.apiKey["api_access_token"] = "USER_API_KEY"
        // ApiClient.apiKey["api_access_token"] = "AGENT_BOT_API_KEY"

        val updateConversationRequest = UpdateConversationRequest()

        try
        {
            ConversationsApi().updateConversation(
                accountId = 0,
                conversationId = 0,
                _data = updateConversationRequest,
            )
        } catch (e: ClientException) {
            println("4xx response calling ConversationsApi#updateConversation")
            e.printStackTrace()
        } catch (e: ServerException) {
            println("5xx response calling ConversationsApi#updateConversation")
            e.printStackTrace()
        }
    }
}
